src/page/admin/auth/AutoCreate/AutoCreateForm.jsx

```diff
@@ -0,0 +1,96 @@
+        setValue('defaultRootRole', evt.target.value);
+    };
+
+    const updateField = e => {
+        setValue(e.target.name, e.target.value);
+    };
+
+return (
+    <Fragment>
+        <Grid container spacing={3}>
+            <Grid item md={5}>
+                <strong>Auto-create users</strong>
+                <p>
+                    Enable automatic creation of new users when signing in.
+                </p>
+            </Grid>
+            <Grid item md={6} style={{ padding: '20px' }}>
+                <Switch
+                    onChange={updateAutoCreate}
+                    name="enabled"
+                    checked={data.autoCreate}
+                    disabled={!data.enabled}
+                >
+                    Auto-create users
+                </Switch>
+            </Grid>
+        </Grid>
+        <Grid container spacing={3}>
+            <Grid item md={5}>
+                <strong>Default Root Role</strong>
+                <p>
+                    Choose which root role the user should get when no explicit role mapping exists.
+                </p>
+            </Grid>
+            <Grid item md={6}>
+            <FormControl style={{minWidth: '200px'}}>
+                <InputLabel id="defaultRootRole-label">Default Role</InputLabel>
+                <Select
+                    labelId="defaultRootRole-label"
+                    id="defaultRootRole"
+                    name="defaultRootRole"
+                    disabled={!data.autoCreate}
+                    value={data.defaultRootRole || 'Editor'}
+                    onChange={updateDefaultRootRole}
+                    >
+                    {/*consider these from API or constants. */}
+                    <MenuItem value='Viewer'>Viewer</MenuItem>
+                    <MenuItem value='Editor'>Editor</MenuItem>
+                    <MenuItem value='Admin'>Admin</MenuItem>
+                </Select>
+            </FormControl>
+            </Grid>
+        </Grid>
+        <Grid container spacing={3}>
+            <Grid item md={5}>
+                <strong>Email domains</strong>
+                <p>
+                    Comma separated list of email domains
+                    that should be allowed to sign in.
+                </p>
+            </Grid>
+            <Grid item md={6}>
+                <TextField
+                    onChange={updateField}
+                    label="Email domains"
+                    name="emailDomains"
+                    disabled={!data.autoCreate}
+                    required={!!data.autoCreate}
+                    value={data.emailDomains || ''}
+                    placeholder="@company.com, @anotherCompany.com"
+                    style={{ width: '400px' }}
+                    rows={2}
+                    variant="outlined"
+                    size="small"
+                />
+            </Grid>
+        </Grid>
+    </Fragment>);
+}
+
+AutoCreateForm.propTypes = {
```

Comment:
Consider creating new files in TS to avoid adding code needs to be reworked in the future.

Reply:
Agreed. Would '.tsx' ending be preferred?

Reply:
> Agreed. Would '.tsx' ending be preferred?

Yes, that is the convention for the rest of the typescript components we have in this project 👍

Reply:
jikes. the event types required me to look up some docs 🤣